crates/nargo/src/package.rs

```diff
@@ -0,0 +1,36 @@
+use std::{collections::BTreeMap, path::PathBuf};
+
+use noirc_frontend::graph::{CrateName, CrateType};
+
+use crate::{
+    constants::{PROVER_INPUT_FILE, VERIFIER_INPUT_FILE},
+    workspace::Workspace,
+};
+
+#[derive(Clone)]
+pub enum Dependency {
+    Local { entry_path: PathBuf, crate_type: CrateType, workspace: Workspace },
+    Remote { entry_path: PathBuf, crate_type: CrateType, workspace: Workspace },
+}
+
+#[derive(Clone)]
+pub struct Package {
+    pub root_dir: PathBuf,
+    pub crate_type: CrateType,
+    pub entry_path: PathBuf,
+    pub name: Option<CrateName>,
```

Comment:
Why would name be optional?

Reply:
This is the current implementation, but I think it's wrong and we should require names in crates.

Reply:
Done in #2056 and rebased against.